src/controllers/plugins/Plugin.hpp

```diff
@@ -0,0 +1,158 @@
+#pragma once
+#ifdef CHATTERINO_HAVE_PLUGINS
+#    include "Application.hpp"
+#    include "controllers/commands/CommandController.hpp"
+
+#    include <magic_enum.hpp>
+#    include <QDir>
+#    include <QJsonArray>
+#    include <QJsonObject>
+#    include <QString>
+#    include <semver/semver.hpp>
+
+#    include <set>
+#    include <vector>
+
+struct lua_State;
+
+namespace chatterino {
+
+struct PluginMeta {
```

Comment:
Could use some comments in this, even if they just point back to the json schema maybe